core/unit_test/TestFunctorAnalysis.hpp

```diff
@@ -85,7 +85,7 @@ void test_functor_analysis() {
   TestFunctorAnalysis_03 c03;
   using A03 = Kokkos::Impl::FunctorAnalysis<
       Kokkos::Impl::FunctorPatternInterface::REDUCE,
-      Kokkos::RangePolicy<ExecSpace>, TestFunctorAnalysis_03>;
+      Kokkos::RangePolicy<ExecSpace>, TestFunctorAnalysis_03, void>;
```

Comment:
Should a non-void `OverrideValueType` test be added where we static_assert the value type is the override?

Reply:
Sure. I added another test.